server/api/api/endpoints/functions_v2.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2024 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import http
+
+import fastapi
+from fastapi import (
+    APIRouter,
+    Depends,
+    Request,
+)
+from fastapi.concurrency import run_in_threadpool
+from sqlalchemy.orm import Session
+
+import mlrun.common.model_monitoring
+import mlrun.common.model_monitoring.helpers
+import mlrun.common.schemas
+import server.api.api.utils
+import server.api.crud.model_monitoring.deployment
+import server.api.crud.runtimes.nuclio.function
+import server.api.db.session
+import server.api.launcher
+import server.api.utils.auth.verifier
+import server.api.utils.background_tasks
+import server.api.utils.clients.chief
+import server.api.utils.functions
+import server.api.utils.pagination
+import server.api.utils.singletons.k8s
+import server.api.utils.singletons.project_member
+from mlrun.utils import logger
+from server.api.api import deps
+from server.api.utils.singletons.scheduler import get_scheduler
+
+router = APIRouter()
+
+
+@router.delete(
+    "/projects/{project}/functions/{name}",
+    responses={
+        http.HTTPStatus.NO_CONTENT.value: {},
```

Comment:
In what flow is `NO_CONTENT` returned?

Reply:
@TomerShor the only flow we have is creating a background task. So, in any case

Reply:
@TomerShor good point! removed it